### PR TITLE
add new continer to Makefile

### LINKED ISSUE #715: Makefile needs to have img build consolidated into a single target
> /kind enhancement

Currently we have to deal with a container img for jekyll, yaspeller, yaspeller, htmlprooder and casperjs.  Let's combine all these pieces into a single img.  This will make CI much easier to deal with.

### PR DESCRIPTION
**Pull request by eifrach**
Signed-off-by: Eran Ifrach <eifrach@redhat.com>

**What this PR does **:

* Modifying Make filewith a single container for all tests/runs
* Depends on [PR-1115](https://github.com/kubevirt/project-infra/pull/1155) on Sysinfra 

> Fixes #715

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,87 +158,21 @@ check_links: | envvar stop
...
-	else \
-		IMAGE="`echo $${TAG} | sed -e s#\'##g -e s#localhost\/## -e s#:latest##`"; \
-		echo "DOCKERFILE file: ./Dockerfile"; \
-		echo "Be sure to add changes to upstream: kubevirt/project-infra/master/images/$${IMAGE}/Dockerfile"; \
-		echo; \
-	fi; \
-	${CONTAINER_ENGINE} rmi ${TAG} 2> /dev/null || echo -n; \
-	${BUILD_ENGINE} ${TAG}; \
-	if [ "$${REMOTE}" ]; then rm -f Dockerfile > /dev/null 2>&1; fi
+	${DEBUG} curl $${REPOPATH}/update-yaspeller.sh | bash -  > /dev/null  2>&1 || echo "${YELLOW} Cannot get latest yaspeller, Working with local yaspeller ${RESET}"
```
Comment (mazzystr) [inline Makefile:161]: How about we skip this part so we don't have to rely on the project-infra guys and their CI?

**Review by mazzystr — CHANGES_REQUESTED** [review]

**Review by mazzystr — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -86,17 +86,51 @@ endif
...
+ifeq ($(shell test $IMGTAG > /dev/null 2>&1 && printf "true"), true)
+	@echo WARN: Using IMGTAG=$$IMGTAG
+	@echo
+else
+	@$(eval export IMGTAG=localhost/kubevirt-kubevirt.github.io)
+endif
+endif
+
+
+## Build image localhost/kubevirt.io
```
Comment (mazzystr) [inline Makefile:105]: localhost/kubevirt-kubevirt.github.io

**Review by mazzystr — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -124,87 +158,21 @@ check_links: | envvar stop
...
-		IMAGE="`echo $${TAG} | sed -e s#\'##g -e s#localhost\/## -e s#:latest##`"; \
-		echo "DOCKERFILE file: ./Dockerfile"; \
-		echo "Be sure to add changes to upstream: kubevirt/project-infra/master/images/$${IMAGE}/Dockerfile"; \
-		echo; \
-	fi; \
-	${CONTAINER_ENGINE} rmi ${TAG} 2> /dev/null || echo -n; \
-	${BUILD_ENGINE} ${TAG}; \
-	if [ "$${REMOTE}" ]; then rm -f Dockerfile > /dev/null 2>&1; fi
+	${DEBUG} curl $${REPOPATH}/update-yaspeller.sh | bash -  > /dev/null  2>&1 || echo "${YELLOW} Cannot get latest yaspeller, Working with local yaspeller ${RESET}"
+	${CONTAINER_ENGINE} run -it --rm --name yaspeller -v ${PWD}:/srv:ro${SELINUX_ENABLED} -v /dev/null:/srv/Gemfile.lock --workdir=/ ${IMGTAG} /bin/bash -c 'yaspeller -c /srv/.yaspeller.json --only-errors --ignore-tags iframe,img,code,kbd,object,samp,script,style,var /srv'
```
Comment (mazzystr) [inline Makefile:162]: change `--name yaspeller` to `--name website`

**Review by mazzystr — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -124,87 +158,21 @@ check_links: | envvar stop
...
 	for i in .jekyll-cache _site Gemfile.lock; do rm -rf ./"$${i}" 2> /dev/null; echo -n; done
-	${CONTAINER_ENGINE} run -d --name website --net=host -v ${PWD}:/srv/jekyll:ro${SELINUX_ENABLED} -v /dev/null:/srv/jekyll/Gemfile.lock --mount type=tmpfs,destination=/srv/jekyll/_site --mount type=tmpfs,destination=/srv/jekyll/.jekyll-cache jekyll/jekyll /bin/bash -c "jekyll serve --trace --force_polling --future"
+	${CONTAINER_ENGINE} run -d --name website --net=host -v ${PWD}:/srv/jekyll:ro${SELINUX_ENABLED} -v /dev/null:/srv/jekyll/Gemfile.lock --mount type=tmpfs,destination=/srv/jekyll/_site --mount type=tmpfs,destination=/srv/jekyll/.jekyll-cache --workdir=/srv/jekyll ${IMGTAG} /bin/bash -c "jekyll serve --host=0.0.0.0 --trace --force_polling --future"
 	@echo
 
 
 ## Container status
 status: | envvar
-	@echo "${GREEN}Makefile: Check image status${RESET}"
+	@echo "${GREEN}Makefile: Check Container status${RESET}"
```
Comment (mazzystr) [inline Makefile:175]: `Container` should be `container`

File: Makefile (hunk 1 above)
Comment (eifrach) [reply Makefile:161]: we are replaying on it with the docker file.
but i can push it in to this repo

File: Makefile (hunk 2 above)
Comment (eifrach) [reply Makefile:105]: adding change

File: Makefile (hunk 3 above)
Comment (eifrach) [reply Makefile:162]: adding change

File: Makefile (hunk 4 above)
Comment (eifrach) [reply Makefile:175]: adding change

File: Makefile [hunk 5]
```diff
@@ -214,16 +182,3 @@ stop: | envvar
 	@echo "${GREEN}Makefile: Stop site${RESET}"
```
[the remark this answers is not in the document]
Comment (eifrach) [reply Makefile:182]: adding change

File: Makefile (hunk 1 above)
Comment (eifrach) [reply Makefile]: added changes

**Comment by mazzystr** [comment]
@eifrach This is a much cleaner pull req.  Thanks for recreating it.  

Since we're changing the way the Makefile works we need to update the README.md to reflect those changes.  Please add a commit for that.

Then all there is just a few minor changes that need fixed.


More on L161 .... I'm gonna insist that the old script get put back.  I had to do locally in order to test `make check_spelling` since otherwise I can't test until the CI team takes a look and merges pull/115.  In the future if we want to expand the script we have to check it in to project repo ... then test here.  That's a pretty bad dependency.  We also have to bolt on additional script functions such as `|` and `||` to get around file system permissions and give a problem notice to the user.  We're in the exact same spot as before with additional complexity and risk.

If you put the old script back we can merge this pretty quick when the minors get fixed.  

We can talk about consolidating all the file retrieve code in another issue/pull_req.

**Comment by eifrach** [comment]
@mazzystr all changes has been added

**Pull request merged**